src/solvers/smtlib/CMakeLists.txt

```diff
@@ -21,7 +21,11 @@ target_include_directories(smtlib
 # could be very large and the std::list destructors use a lot of stack space
 set(stacksize 20971520)  # 20 MB ought to be enough for everyone
 if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /STACK:${stacksize}" CACHE STRING "Linker flags" FORCE)
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /STACK:${stacksize}" CACHE STRING "Linker flags" FORCE)
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
```

Comment:
Are you sure we should do this for Clang regardless of whether it's Apple's or not?

Clang supports different linkers via `-fuse-ld=(bfd|lld|...)`, not sure about the Apple one. Unfortunately in https://gitlab.kitware.com/cmake/cmake/-/issues/18209 they say CMake does not and probably will not support something like CMAKE_LINKER_ID. Instead they suggest to use `try_compile()`.